message_definitions/v1.0/development.xml

```diff
@@ -38,5 +47,36 @@
       <field type="float" name="raw_press" units="hPa">Raw differential pressure. NaN for value unknown/not supplied.</field>
       <field type="uint8_t" name="flags" enum="AIRSPEED_SENSOR_FLAGS">Airspeed sensor flags.</field>
     </message>
+    <message id="420" name="RADIO_RC_CHANNELS">
+      <description>RC channel outputs from a MAVLink RC receiver for input to a flight controller or other components
+        (allows an RC receiver to connect via MAVLink instead of some other protocol such as PPM-Sum or S.BUS).
+        Note that this is not intended to be an over-the-air format, and does not replace RC_CHANNELS and similar
+        messages reported by the flight controller. The target_system field should normally be set to the system id of
+        the system to control, typically the flight controller. The target_component field can normally be set to 0, so
+        that all components of the system can receive the message.
+        The channels array field can publish up to 32 channels; the number of channel items used in the array is
+        specified in the count field.
+        The time_last_update_ms field contains the timestamp of the last received valid channels data in the
+        receiver's time domain, and the channel items up to count holds the last valid data.
+        If the channels data are not up-to-date and do not hold the current/latest data, the RADIO_RC_CHANNELS_FLAGS_OUTDATED
+        flag is set by the receiver.
+        The RADIO_RC_CHANNELS_FLAGS_FAILSAFE failsafe flag is set by the receiver if the receiver's failsafe condition
+        is met (implementation dependent, e.g., connection to the RC radio is lost). In this case time_last_update_ms
+        still contains the timestamp of the last valid channels data, but the content of the channels data is not
+        defined by the protocol but is up to the implementation of the receiver. For instance, the channels data could
+        contain failsafe values configured in the receiver; the default is to carry the last valid data.
+        Note: The RC channels fields are extensions to ensure that they are located at the end of the serialized payload
+        and subject to MAVLink's trailing-zero trimming.
+      </description>
```

Comment:
This makes relatively minor modifications for English, not for the behaviour.

1. Sentence line breaks - make it easier to read (for me) than arbitrary line breaks.
2. Reworded the text around RADIO_RC_CHANNELS_FLAGS_OUTDATED - meaning is the same.
3. Restructured to say "The count field indicates the first index of the channel array that holds invalid data.". I've assumed MAVLink uses array indexing from 0 but perhaps we should never assume and just say "this is the count of items that are used" because different programming languages take different conventions.

```suggestion
      <description>RC channel outputs from a MAVLink RC receiver for input to a flight controller or other components (allows an RC receiver to connect via MAVLink instead of some other protocol such as PPM-Sum or S.BUS).
        Note that this is not intended to be an over-the-air format, and does not replace RC_CHANNELS and similar
        messages reported by the flight controller.
        The target_system field should normally be set to the system id of the system to control, typically the flight controller.
        The target_component field can normally be set to 0, so that all components of the system can receive the message.
        The channels array field can publish up to 32 channels; the number of channel items used in the array is specified in the count field.
        The time_last_update_ms field contains the timestamp of the last received valid channels data in the receiver's time domain.
        The count field indicates the first index of the channel array that holds invalid data.
        The RADIO_RC_CHANNELS_FLAGS_OUTDATED is set by the receiver if the channels data is not up-to-date (for example, if new data from the transmitter could not be decoded so the previous data is resent).
        The RADIO_RC_CHANNELS_FLAGS_FAILSAFE failsafe flag is set by the receiver if the receiver's failsafe condition is met (implementation dependent, e.g., connection to the RC radio is lost).
        In this case time_last_update_ms still contains the timestamp of the last valid channels data, but the content of the channels data is not defined by the protocol (it is up to the implementation of the receiver).
        For instance, the channels data could contain failsafe values configured in the receiver; the default is to carry the last valid data.
        Note: The RC channels fields are extensions to ensure that they are located at the end of the serialized payload
        and subject to MAVLink's trailing-zero trimming.
      </description>
```